nft_ingester/src/api/dapi/response.rs

```diff
@@ -75,6 +79,68 @@ impl From<AssetSignature> for SignatureItem {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default, JsonSchema)]
+#[serde(default)]
+pub struct MasterAssetEditionsInfoModel {
+    pub master_edition_address: String,
+    pub supply: u64,
+    pub max_supply: Option<u64>,
+    pub editions: Vec<AssetEditionInfoModel>,
+
+    #[serde(flatten)]
+    pub pagination: PaginationModel,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default, JsonSchema)]
+pub struct AssetEditionInfoModel {
+    pub mint: String,
+    pub edition_address: String,
+    pub edition: u64,
+}
+impl From<&AssetEditionInfo> for AssetEditionInfoModel {
+    fn from(value: &AssetEditionInfo) -> Self {
+        Self {
+            mint: value.mint.to_string(),
+            edition_address: value.edition_address.to_string(),
+            edition: value.edition,
+        }
+    }
+}
+
+impl From<(MasterAssetEditionsInfo, Pagination)> for MasterAssetEditionsInfoModel {
+    fn from((value, pagination): (MasterAssetEditionsInfo, Pagination)) -> Self {
+        Self {
+            master_edition_address: value.master_edition_address.to_string(),
+            supply: value.supply,
+            max_supply: value.max_supply,
+            editions: value.editions.iter().map(AssetEditionInfoModel::from).collect(),
+            pagination: PaginationModel {
+                total: value.editions.len() as u32,
```

Comment:
are 100% sure such a conversion is safe?

Reply:
i think solana just wouldn't allow to store more (in byte value) than u32::MAX of editions

Reply:
replaced on u32::try_from